runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6754,6 +6779,16 @@ TR_J9ByteCodeIlGenerator::genNewArray(int32_t typeIndex)
         }
      }
 
+   // Special case for handling Arrays.copyOfRangeByte when called from StringBuilder.tostring method
+   // The call chain is: StringBuilder.toString() -> String<init>(StringBuilder) -> String<init>(AbstractStringBuilder, Void)
+   //                                                             -> Arrays.copyOfRange([BII) -> Arrays.copyOfRangeByte([BII)
+   if ((comp()->getInlineDepth() >= 4) && !comp()->isPeekingMethod() && !generateArraylets
```

Comment:
I think a check to see the callerIndex of `copyOfRangeByte` is >= 2 can be done in `isNewArrayResultOfStringBuilderToString` .